src/system/win32-wgl/resource.c

```diff
@@ -0,0 +1,129 @@
+    0x1DF101D3, 0x17AE4CA6, 0x094F9B39, 0x0310D64C,
+    0x0F6E37A0, 0x05317AD5, 0x1BD0AD4A, 0x118FE03F, // 0xC0 - 0xCF
+    0x1BABDCF7, 0x11F49182, 0x0F15461D, 0x054A0B68,
+    0x1B5D3F8D, 0x110272F8, 0x0FE3A567, 0x05BCE812,
+    0x0F98D4DA, 0x05C799AF, 0x1B264E30, 0x11790345,
+    0x1AB0F979, 0x10EFB40C, 0x0E0E6393, 0x04512EE6, // 0xD0 - 0xDF
+    0x0E75122E, 0x042A5F5B, 0x1ACB88C4, 0x1094C5B1,
+    0x0E83F154, 0x04DCBC21, 0x1A3D6BBE, 0x106226CB,
+    0x1A461A03, 0x10195776, 0x0EF880E9, 0x04A7CD9C,
+    0x196B7491, 0x133439E4, 0x0DD5EE7B, 0x078AA30E, // 0xE0 - 0xEF
+    0x0DAE9FC6, 0x07F1D2B3, 0x1910052C, 0x134F4859,
+    0x0D587CBC, 0x070731C9, 0x19E6E656, 0x13B9AB23,
+    0x199D97EB, 0x13C2DA9E, 0x0D230D01, 0x077C4074,
+    0x0CB5BA48, 0x06EAF73D, 0x180B20A2, 0x12546DD7, // 0xF0 - 0xFF
+    0x1870511F, 0x122F1C6A, 0x0CCECBF5, 0x06918680,
+    0x1886B265, 0x12D9FF10, 0x0C38288F, 0x066765FA,
+    0x0C435932, 0x061C1447, 0x18FDC3D8, 0x12A28EAD
+};
+
+int breadbox_resource_load(breadbox_resource_t *res, const char *id) {
+    FILE *bbr;
+    void *data;
+    breadbox_resource_header_t header;
+    size_t length;
+    // I'm using the filesystem to load resources for now. Just keep in mind
+    // that this won't be the case in the final version of the engine! ~Alex
+    if(bbr = fopen(id, "r")) {
+        fseek(bbr, 0, SEEK_END);
+        if((length = ftell(bbr)) > sizeof(breadbox_resource_header_t)) {
+            fseek(bbr, 0, SEEK_SET);
+            if(fread(&header, sizeof(breadbox_resource_header_t), 1, bbr)) {
+                // *Insert whole rant about endianness here*
+                // See resources.h for details. ~Alex
+                if(header.signature == 0x00524242) {
+                    if(header.platform == BBPLAT_NEUTRAL || header.platform == BBPLAT_UNIX_GLX) {
+                        if(data = malloc(header.size)) {
+                            if(fread(data, 1, header.size, bbr)) {
+                                res->data = data;
+                                res->hash = header.hash;
+                                res->size = header.size;
+                                res->type = header.type;
+                                if(!breadbox_resource_verify(res)) {
+                                    fclose(bbr);
+                                    return 0;
+                                }
+                                res->data = NULL;
+                            }
+                            free(data);
+                        }
+                    }
+                }
+            }
+        }
+        fclose(bbr);
+    }
+    return 1;
+}
+
+void breadbox_resource_unload(breadbox_resource_t *res) {
+    free(res->data);
+    res->data = NULL;
+}
+
+int breadbox_resource_verify(breadbox_resource_t *res) {
+    uint32_t check = 0xFFFFFFFF;
+    for(int i = 0; i < res->size; i++) {
+        check = CRC32C_TABLE[(char)check ^ ((char *)res->data)[i]] ^ (check >> 8);
```

Comment:
This somehow calculates the incorrect hash differently from the incorrect hash generated by bbconv. The table needs to be fixed before we can determine which version is incorrect.